xarray/core/duck_array_ops.py

```diff
@@ -239,23 +262,24 @@ def as_shared_dtype(scalars_or_arrays, xp=np):
         ):
             return scalars_or_arrays
         raise ValueError(
-            f"Cannot cast arrays to shared type, found array types {[x.dtype for x in scalars_or_arrays]}"
+            "Cannot cast arrays to shared type, found"
+            f" array types {[x.dtype for x in scalars_or_arrays]}"
         )
 
     # Avoid calling array_type("cupy") repeatidely in the any check
     array_type_cupy = array_type("cupy")
     if any(isinstance(x, array_type_cupy) for x in scalars_or_arrays):
         import cupy as cp
 
-        arrays = [asarray(x, xp=cp) for x in scalars_or_arrays]
-    else:
-        arrays = [asarray(x, xp=xp) for x in scalars_or_arrays]
+        xp = cp
```

Comment:
Is this code even necessary with the new `get_array_namespace()` that checks multiple arrays? Consider removing it if possible.

Reply:
actually, no. `cupy` does not appear to define `__array_namespace__` yet, so that wouldn't help us. I guess we'll have to keep the check.